tests/common/include/tests/common/grid_grid_collection.inl

```diff
@@ -0,0 +1,126 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+// Detray include(s)
+#include "detray/definitions/indexing.hpp"
+#include "detray/masks/cylinder3D.hpp"
+#include "detray/surface_finders/grid/axis.hpp"
+#include "detray/surface_finders/grid/grid.hpp"
+#include "detray/surface_finders/grid/grid_builder.hpp"
+#include "detray/surface_finders/grid/grid_collection.hpp"
+#include "detray/surface_finders/grid/populator.hpp"
+#include "detray/surface_finders/grid/serializer.hpp"
+
+// System include(s)
+#include <algorithm>
+#include <climits>
+
+using namespace detray;
+using namespace detray::n_axis;
+
+namespace {
+
+// non-owning multi-axis: Takes external containers
+bool constexpr is_owning = true;
+bool constexpr is_n_owning = false;
+
+constexpr dindex inf{std::numeric_limits<dindex>::max()};
+
+// Create some bin data for non-owning grid
+template <class populator_t, typename entry_t>
+struct bin_content_sequence {
+
+    entry_t entry{0};
+
+    auto operator()() {
+        entry += entry_t{1};
+        return populator_t::init(entry);
+    }
+};
+
+}  // anonymous namespace
+
+/// Unittest: Test the construction of a collection of grids
+TEST(grid, grid_collection) {
+
+    // grid type
+
+    // Non-owning grid type with array<dindex, 3> as bin content
+    using cylindrical_3D =
+        coordinate_axes<cylinder3D::axes<>, is_n_owning, host_container_types>;
+    using grid_t =
+        grid<cylindrical_3D, dindex, simple_serializer, regular_attacher<3>>;
+
+    // Build test data
+
+    // Offsets into edges container and #bins for all axes
+    dvector<dindex_range> edge_ranges = {{0u, 2u},  {2u, 4u},  {4u, 6u},
+                                         {6u, 1u},  {8u, 3u},  {10u, 8u},
+                                         {12u, 5u}, {14u, 5u}, {16u, 5u}};
+
+    // Bin edges for all axes
+    dvector<scalar> bin_edges = {-10, 10., -20., 20., 0.,  120., -5., 5., -15.,
```

Comment:
What are the edge ranges and bin edges exactly?  Edge is something like the boundary of a axis bin?

Reply:
Thanks for the detailed answer. Could you check if I understand the test codes correctly based on your explanation?
BTW, I still don't get what the offsets into edge container mean.

```
edge_ranges = {{axis1 offset in edge container, # of bins of axis1}, ..., {axis9 offset in edge container, # of bins of axis9}}
bin_edges = { min of axis1, max of axis1, min of axis2, max of axis2, ..., min of axis9, max of axis9 }

{ axis1 (r), axis2 (phi), axis3 (z) } -> grid1
{ axis4 (r), axis5 (phi), axis6 (z) } -> grid2
{ axis7 (r), axis8 (phi), axis9 (z) } -> grid3
```

Reply:
Also, all grids should be instantiated in the grid collection constructor? It is not possible to append a grid one-by-one like we push back an element in vector?

Reply:
That looks correct. Imagine axis2 has irregular binning and instead of holding ```min of axis2, max of axis2``` in the bin edges container, it holds an arbitrary number of bin edges. How would axis3 know where to look for its span? That's what I need the offset for, unfortunately.

Reply:
You are right, grid collections should not be constructed like this in the detector building code. Sadly it not so easy to just add a ```push_back```, since the grids that hold data have a different type from the non-owning grids in the grid collection. I am working on a grid builder right know to see how to solve this exact issue